### DISCUSSION TITLE
PHP HTML EMAILS NOT WORKING

### DISCUSSION THREAD

**Discussion by wtimu** [community/community#122684] [Programming Help] [answered]
### Body

Hello, I have a form for sending emails in my html project linked to a php file in my repository, but when i try to send an email, it returns 

> 405 Not Allowed.

How do I solve that issue. Attached are my html form and php code

**HTML form found in _Adtim/contact.html_**

```
<form action="PHP/submit_feedback.php" method="post">
    <div>
        <label for="name"></label>
        <input type="text" placeholder="Name" id="name" name="name" required>
    </div>
    <div>
        <label for="phone"></label>
        <input type="text" placeholder="Phone Number" id="phone" name="phone">
    </div>
    <div>
        <label for="email"></label>
        <input type="email" placeholder="Email" id="email" name="email" required>
    </div>
    <div>
        <label for="message"></label>
        
		<input type="text" class="message-box" placeholder="Message" id="message" name="message" required />
    </div>
    <div class="d-flex">
        <button type="submit">SEND</button>
    </div>
</form>
```

**Here is the PHP Code found in _Adtim/PHP/submit_feedback.php_**

```
<?php
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    // Get the form fields and remove whitespace
    $name = strip_tags(trim($_POST["name"]));
    $phone = strip_tags(trim($_POST["phone"]));
    $email = filter_var(trim($_POST["email"]), FILTER_SANITIZE_EMAIL);
    $message = trim($_POST["message"]);

    // Check for empty fields
    if (empty($name) || empty($email) || empty($message)) {
        // Set a 400 (bad request) response code and exit
        http_response_code(400);
        echo "Please fill out all required fields.";
        exit;
    }

    // Set the recipient email address
    $recipient = "support@adtimtech.com";

    // Set the email subject
    $subject = "New message from $name";

    // Build the email content
    $email_content = "Name: $name\n";
    $email_content .= "Phone: $phone\n";
    $email_content .= "Email: $email\n\n";
    $email_content .= "Message:\n$message\n";

    // Build the email headers
    $email_headers = "From: $name <$email>";

    // Send the email
    if (mail($recipient, $subject, $email_content, $email_headers)) {
        // Set a 200 (okay) response code
        http_response_code(200);
        echo "Thank you! Your message has been sent.";
    } else {
        // Set a 500 (internal server error) response code
        http_response_code(500);
        echo "Oops! Something went wrong and we couldn't send your message.";
    }
} else {
    // Not a POST request, set a 403 (forbidden) response code
    http_response_code(403);
    echo "There was a problem with your submission, please try again.";
}
?>

```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by aroonmircam** [comment]
Make sure that the submit_feedback.php file is located in the correct path and is accessible from the path specified in the action attribute of the HTML form.

**Reply by wtimu** [reply]
It is in the correct path. I actually just copied the path into my code

**Comment by liker912** [comment]
Hi, I think needs to check the next things:

1. Server configuration: Ensure that your web server is properly configured to handle POST requests for the specified URL. Check the configuration files for any rules that might block or redirect POST requests.

2. If the web server has the necessary permissions to access the script files and that the scripts are located in the correct directory.

3. Redirection rules can cause the request to be sent using the wrong method. Make sure there are no redirections that could alter the request method from POST to something else.

Also, check logs from your web server. It can help understand why it's doesn't work

**Reply by wtimu** [reply]
I thought github pages handles the what would be my webserver role

**Reply by liker912** [reply, marked as answer]
Github pages works only with a static content and it doesn't support PHP

**Comment by f9ki3** [comment]
Check if any errors are popping up in your PHP script. You might find some clues in your server's error logs. Or maybe the PHP script is using the mail() function correctly. Check if you've got everything set up right - like the recipient, subject, message, and headers. Please also check SMTP Email that you are using if you config it correctly

**Comment by DarioChiappello** [comment]
You must upload your project to a server that supports PHP and has the SMTP port enabled. If you dont have this port enable, it won't work
